CocktailRecipes/CocktailRecipes/Networking/Core/NetworkManager.swift

```diff
@@ -11,59 +11,43 @@ import Foundation
 enum NetworkError: String, Error {
 	case missingURL = "URL is nil."
 	case parameterEncodingFailed = "Parameter encoding failed."
+    case unknown
 }
 
 final class NetworkManager {
-	static var shared = NetworkManager()
 	private let apiKey: String
-	private let session = URLSession.shared
-
-	private init() {
-		guard let path = Bundle.main.path(forResource: "APIKey", ofType: "plist"),
-			let xml = FileManager.default.contents(atPath: path) else { fatalError("APIKey.plist not found") }
-		let apiKey = try? JSONDecoder().decode(String.self, from: xml)
-		self.apiKey = apiKey ?? "1"
-	}
-
-	func request<T: Decodable>(_ endpoint: EndpointConstructable, completion: @escaping (Result<[T], Error>) -> Void) {
-		do {
-			let request = try buildRequest(from: endpoint)
-			session.dataTask(with: request) { (data, response, error) in
-				guard error == nil else {
-					completion(.failure(error!))
-					return
-				}
-				if let data = data {
-					do {
-						let models = try JSONDecoder().decode([T].self, from: data)
-						completion(.success(models))
-					} catch let error {
-						completion(.failure(error))
-					}
-				}
-			}
-		} catch let error {
-			completion(.failure(error))
-		}
-	}
+    private let session: URLSession
+    
+    init(session: URLSession = URLSession.shared) {
```

Comment:
I'm okay with removing this as a singleton but can't we just have line 19 `private let session = URLSession.shared`. I feel like that init parameter is not really necessary since we know what the value is always going to be.

Reply:
We need a way to pass in the MockSession. And this also leaves the NetworkManager open to use other kind of session.

Reply:
yeah I noticed, in the tests you need this parameter to pass the mock session